### ISSUE TITLE
Sample noise flow add to SIDD raw image problem using the pre-train model provided

### ISSUE THREAD

**Issue by circlehy** [BorealisAI/noise_flow#3]
Sample noise flow added to SIDD clean image in raw format, then transfer to RGB format. The added noise seems 'larger' than expected since the parameters were known. I am not sure how to solve the problem.
cam: 0, iso: 100
clean 
![0051_002_S6_00100_00060_5500_N0051_GT_RAW_010_add_0_100 0_0_100_gt_sRGB](https://user-images.githubusercontent.com/9382038/66566995-5454f400-eb77-11e9-92b9-6f17fb6e71aa.png)

noisy
![0051_002_S6_00100_00060_5500_N0051_GT_RAW_010_add_0_100 0_0_100_sRGB](https://user-images.githubusercontent.com/9382038/66567029-6cc50e80-eb77-11e9-8ccc-a90fedc4d8e3.png)

**Comment by jmu201521121021** [comment]
@circlehyhello,  how do you transfer Raw format to RGB format?

**Comment by circlehy** [comment]
> @circlehyhello, how do you transfer Raw format to RGB format?

They provided the transfer code on the SIDD website.

**Comment by jmu201521121021** [comment]
@circlehy thank you!

**Comment by jmu201521121021** [comment]
@circlehy "The added noise seems 'larger' than expected since the parameters were known",I also encountered this problem, do you solve the problem?

**Comment by Fanbenchao** [comment]
I also encountered this problem. I'll appreciate it that the authors could answer our questions.
cam: GP
iso: 3200
clean:
![0](https://user-images.githubusercontent.com/30153284/68473084-035f1b00-025d-11ea-82b9-8115253c08e2.jpg)
noise:
![0](https://user-images.githubusercontent.com/30153284/68473130-14a82780-025d-11ea-87fa-06a3d0db899a.jpg)
generate:
![0](https://user-images.githubusercontent.com/30153284/68473141-1eca2600-025d-11ea-9954-3b753390214d.jpg)

**Comment by circlehy** [comment]
> @circlehy "The added noise seems 'larger' than expected since the parameters were known",I also encountered this problem, do you solve the problem?

No, the released model seems not work.

**Comment by AbdoKamel** [comment]
Hi, 

There was a bug in the sampling code, it was not using the trained batch norm layers, this is fixed now. Also, I have added code for rendering the sampled noisy images into sRGB.

**Comment by jmu201521121021** [comment]
nice,thank you,
发自我的iPhone

------------------ Original ------------------

**Comment by JimmyChame** [comment]
Hi, I still find some problems in  the sampling code. I just use the pre-train model provided to generate some samples. The added noise seems different than expected: (The left is clean image, the center is original noisy image and the right is the generated noisy image)
![10_03_0800](https://user-images.githubusercontent.com/29774815/75523698-71fff880-5a47-11ea-8e9f-f2e9d10d66bd.png)
![10_02_0800](https://user-images.githubusercontent.com/29774815/75524543-16366f00-5a49-11ea-95b4-f4de6487441c.png)

Then I use the generated noisy images to finetune a denoising model pre-trained on SIDD dataset, the PSNR on SIDD validation dataset is decreasing significantly (from 39 down to 37). I am not sure if there are still bugs in the code.

**Comment by SSRheart** [comment]
> Hi, I still find some problems in the sampling code. I just use the pre-train model provided to generate some samples. The added noise seems different than expected: (The left is clean image, the center is original noisy image and the right is the generated noisy image)
> ![10_03_0800](https://user-images.githubusercontent.com/29774815/75523698-71fff880-5a47-11ea-8e9f-f2e9d10d66bd.png)
> ![10_02_0800](https://user-images.githubusercontent.com/29774815/75524543-16366f00-5a49-11ea-95b4-f4de6487441c.png)
> 
> Then I use the generated noisy images to finetune a denoising model pre-trained on SIDD dataset, the PSNR on SIDD validation dataset is decreasing significantly (from 39 down to 37). I am not sure if there are still bugs in the code.

Hi, have you solved this problem？ The same results appear on my trained model.

**Comment by Vandermode** [comment]
Any update on this problem? 
Actually, I find the provided pretrained model is a simplified version of the one described on the paper, since the width parameter is set to 4 instead of 32 mentioned by authors

**Comment by ymcasky** [comment]
![10_02_0800](https://user-images.githubusercontent.com/6229000/85495103-79599c00-b60c-11ea-883a-38496094bc2c.png)
The same problem when run sample_noise_flow.py with "patch_size" set to 480.
Have you solved this problem?

**Comment by AbdoKamel** [comment]
Please check out this potential fix:
https://github.com/BorealisAI/noise_flow/commit/f9e2d29f350f8d59fd6be9af58ab41eeaafeb1db

**Comment by yishionsunshine** [comment]
I hava a question: why the patch height is 32 in your code but described 64 in your paper?
